workshops/templates/reports/training_completion_rates.txt

```diff
@@ -1,6 +1,2 @@
-------------------------------------
-{{ training.slug }} ({{ training.tags.all|join:", " }}), badge: {{ badge }}
-{% with learners_len=learners.count completed_len=completed.count completed_other_len=completed_other.count no_badge_len=no_badge.count %}Learners: {{ learners_len }}
-Completed this training with badge: {{ completed_len }} ({% widthratio completed_len learners_len 100 %}%)
-Completed other training with badge: {{ completed_other_len }} ({% widthratio completed_other_len learners_len 100 %}%)
-No badge: {{ no_badge_len }} ({% widthratio no_badge_len learners_len 100 %}%){% endwith %}
+start,slug,online,badge,learners,completed this,completed this %age,completed other,completed other %age,no badge, no badge %{% for record in records %}
+{{ record.start }},{{ record.training.slug }},{{ record.online }},{{ record.badge }},{{ record.learners_len }},{{ record.completed_len }},{% widthratio record.completed_len record.learners_len 100 %}%,{{ record.completed_other_len }},{% widthratio record.completed_other_len record.learners_len 100 %}%,{{ record.no_badge_len }},{% widthratio record.no_badge_len record.learners_len 100 %}%{% endfor %}
```

Comment:
If you're generating a CSV, then it's better to use Python [`csv`](https://docs.python.org/3/library/csv.html) module than Django templates.